svn log -v -r376544 http://svn.apache.org/repos/asf
------------------------------------------------------------------------
r376544 | jgallacher | 2006-02-09 23:43:55 -0500 (Thu, 09 Feb 2006) | 18 lines
Changed paths:
   M /httpd/mod_python/trunk/src/connobject.c
   M /httpd/mod_python/trunk/src/filterobject.c
   M /httpd/mod_python/trunk/test/httpdconf.py
   M /httpd/mod_python/trunk/test/test.py

Added support for Apache 2.2. Ref MODPYTHON-78

Changed connobject makesockaddr to directly access the address port
rather than using apr_sockaddr_port_get, which was deprecated and removea
d in apr 1.x.

Removed APR_STATUS_IS_SUCCESS which is deprecated. Code which used this
macro will now just compare any return condition with APR_SUCCESS.

Added support to unit test to detect apache version.

Modified unit tests to accomodate changes to mod_auth in apache 2.2.
Mod_auth has been split into multiple modules. The mod_python unit test
will now use mod_auth_basic when apache 2.2 is detected.

An explicit value for KeepAliveTimeout is now set in the unit test when
apache 2.2 is detected.

------------------------------------------------------------------------
Index: src/filterobject.c
===================================================================
--- src/filterobject.c	(revision 376543)
+++ src/filterobject.c	(revision 376544)
@@ -178,7 +178,7 @@
                                   APR_BLOCK_READ, self->readbytes);
         Py_END_ALLOW_THREADS;
 
-        if (!APR_STATUS_IS_EAGAIN(self->rc) && !APR_STATUS_IS_SUCCESS(self->rc)) {
+        if (!APR_STATUS_IS_EAGAIN(self->rc) && !(self->rc == APR_SUCCESS)) {
             PyErr_SetObject(PyExc_IOError, 
                             PyString_FromString("Input filter read error"));
             return NULL;
Index: src/connobject.c
===================================================================
--- src/connobject.c	(revision 376543)
+++ src/connobject.c	(revision 376544)
@@ -79,7 +79,7 @@
         rc = ap_get_brigade(c->input_filters, bb, mode, APR_BLOCK_READ, bufsize);
         Py_END_ALLOW_THREADS;
 
-        if (! APR_STATUS_IS_SUCCESS(rc)) {
+        if (rc != APR_SUCCESS) {
             PyErr_SetObject(PyExc_IOError, 
                             PyString_FromString("Connection read error"));
             return NULL;
@@ -319,14 +319,14 @@
 {
     PyObject *addrobj = makeipaddr(addr);
     PyObject *ret = NULL;
+
+    /* apr_sockaddr_port_get was deprecated and removed in apr 1.x
+     * Access the port directly instead
+     */
     if (addrobj) {
         apr_port_t port;
-        if(apr_sockaddr_port_get(&port, addr)==APR_SUCCESS) {
-            ret = Py_BuildValue("Oi", addrobj, port );
-        }
-        else {
-            PyErr_SetString(PyExc_SystemError,"apr_sockaddr_port_get failure");
-        }
+        port = addr->port; 
+        ret = Py_BuildValue("Oi", addrobj, port );
         Py_DECREF(addrobj);
     }
     return ret;
